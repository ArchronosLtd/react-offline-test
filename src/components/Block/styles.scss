.block {
    position: relative;
    > div {
        border: 1px solid #000;
        height: 30px;

        &:hover {
            background-color: #ccc;
        }
    }
}