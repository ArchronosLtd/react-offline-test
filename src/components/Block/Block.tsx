import React, { Component } from 'react';
import Usage from '../../models/Usage';
import './styles.scss';

type BlockProps = {
    data: Usage
}

// SEE SIMPLE FUNCTION BASED COMPONENTS
export default (props: BlockProps) => {
    return <div className="block">
        <div style={{ width: `${props.data.value}%` }}></div>
        <span>{props.data.label} ({props.data.value}%)</span>
    </div >;
}