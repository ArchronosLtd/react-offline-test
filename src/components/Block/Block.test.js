// Link.react.test.js
import React from 'react';
import Block from './Block';
import renderer from 'react-test-renderer';

describe('Block component', () => {
    const data = {
        id: 'test',
        perc: 50,
        label: 'test'
    };

    it('should have the correct id', () => {
        const component = renderer.create(
            <Block data={data} />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        expect(tree.props.className).toBe('block');
    });
});