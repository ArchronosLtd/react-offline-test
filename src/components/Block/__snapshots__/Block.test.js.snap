// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block component should have the correct id 1`] = `
<div
  className="block"
>
  <div
    style={
      Object {
        "width": "undefined%",
      }
    }
  />
  <span>
    test
     (
    %)
  </span>
</div>
`;
